feat(body): add isTrigger option

age2pierre opened the pull request:
Hello,
I did this small fork to add the "sensor" option to bodies, the original feature-request is referenced here => https://github.com/schteppe/cannon.js/issues/146 and all the credit for the actual code goes to => https://github.com/schteppe/cannon.js/pull/424 ;)
I've committed the `dist/` folder to test without publishing the package, but I can drop it from the PR if needed.

marcofugaro commented:
Thanks for the PR @age2pierre, I think this feature could be useful for many users.

> I've committed the dist/ folder to test without publishing the package, but I can drop it from the PR if needed.

Please exclude it from the PR, otherwise, there will be a lot of conflicts if we build on master 😬 

Some considerations: 

1. I think `isTrigger` would be a better name for it. `sensor` is used in a 2d game engine environment, while 3d game engines use trigger: [triggers in Unity](https://learn.unity.com/tutorial/physics-interactions-colliders-and-triggers-2019-3#5fed83f4edbc2a4dffa6a8d9), [triggers in Unreal](https://docs.unrealengine.com/en-US/Basics/Actors/Triggers/index.html). Bullet calls them [GhostObjects](https://pybullet.org/Bullet/BulletFull/classbtGhostObject.html) 🤷‍♂️.
2. It would be better to have a `trigger.html` example instead of adding the code in the `collisions.html`. The trigger would need to be rendered with a wireframe style, just like [wireframeMaterial](https://github.com/pmndrs/cannon-es/blob/51fa79582eb91a3bc7624163a0cb4afdea71a10b/examples/js/Demo.js#L546) but green or yellow:
```js
this.triggerMaterial = new THREE.MeshBasicMaterial({ color: 0x00ff00, wireframe: true })
``` 
3. The example should show the usage of at least the `beginContact` and `endContact` events. Unity has also the `OnTriggerStay` event, but maybe it's better to do that in a future PR since it's a new event for cannon.

marcofugaro commented on src/objects/Body.ts:
```diff
@@ -246,6 +248,7 @@ export class Body extends EventTarget {
     this.aabbNeedsUpdate = true
     this.boundingRadius = 0
     this.wlambda = new Vec3()
+    this.sensor = options.sensor ?? false
```
it should be a boolean only

```suggestion
    this.sensor = Boolean(options.sensor)
```

age2pierre commented:
@marcofugaro I've updated based upon your review and rebased upon master.
I've kept the demo still very minimal, tell me if you fancy something fancier :)

marcofugaro commented:
Thanks! I adjusted the demo a little bit, opened a new PR with also your commits since I wasn't allowed to edit this one: #83
